core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -488,28 +488,17 @@ open class TileInfo {
         if (terrainFilter == Constants.freshWater && isAdjacentToRiver()) return true
         return (neighbors + this).any { neighbor -> neighbor.matchesFilter(terrainFilter) }
     }
-
+    
```

Comment:
Ha! Trailing whitespace snuck in!
...sorry. 

OK, chosen solution approach is a very simple concept, which is very likely a good thing :tm:. Let's just all go and keep an eye out for any unusual worker behaviour in any playtest. I could still bite myself for not noticing earlier.